ui/perfherder/graphs/GraphsContainer.jsx

```diff
@@ -109,9 +109,9 @@ class GraphsContainer extends React.Component {
       scatterPlotData,
     });
 
-    if (!visibilityChanged) {
```

Comment:
@sarah-clements I can see `visibilityChanged` flag in several files, and seems like a solid logic behind. I tried to change the logic to avoid zooming out to zoomDomain limits on `componentDidMount`, after adding/removing a `legendCard` and after creating an alert from tooltip, but I end up by just removing the flag to get the desired behavior. I ran `yarn test` and nothing seemed to be breaking.
Do you remember what was the logic behind `visibilityChanged`? I can guess, but I can't catch it entirely.
Thanks!

Reply:
It's used to reset the `zoom` query parameter if you add (or remove) another test to the graph. Since it's reset on the graph level we don't want the zoom param to show a mismatch between what's happening on the graph. Maybe it'd be worth adding a comment here to explain that or even changing the variable name so it's clearer.

Reply:
There are 3 places where the `zoom` is reset:
1. when adding/removing a legend card
2. when creating an alert from the tooltip
3. when opening the graph directly from the alert item (the scale from the upper graph - which depends on the datapoints - doesn't the one from the lower graph which is always shown from zero).

1. We (the sheriffs) thought that would be useful when adding/removing a legend card to not reset the zoom. Technically it is possible, but from what you say it might cause some inconsistencies. Is that right?
2. As far as I see, the datapoint are not affected at all by the creation of an alert from tooltip, so I assume this won't be an issue (?)
3. also, we are thinking that when opening the graph from alert item graph link, we could have the scales from the 2 graphs match - like zooming the lower graph to the scale from the upper one. I found a formula, not an exact one, but close enough, that uses the variation interval of the datapoints to leave space upper and lower just as the graph from top.

How does this sound?

Reply:
> 1. We (the sheriffs) thought that would be useful when adding/removing a legend card to not reset the zoom. Technically it is possible, but from what you say it might cause some inconsistencies. Is that right?
> 

If you want to do that, I don't have a problem with it. I was only answering your question about what the `visibilityChanged` logic was for. Feel free to remove it if it's no longer relevant. :)

>3. also, we are thinking that when opening the graph from alert item graph link, we could have the scales from the 2 graphs match - like zooming the lower graph to the scale from the upper one. I found a formula, not an exact one, but close enough, that uses the variation interval of the datapoints to leave space upper and lower just as the graph from top.
> 

Sure.